Extensions/ParticleSystem/particleemitterobject-cocos-renderer.js

```diff
@@ -0,0 +1,291 @@
+    this.renderer.setEmissionRate(objectData.flow);
+
+    this.totalParticles = 1;
+    var that = this;
+    this.renderer.addParticle = function(){
+        cc.ParticleSystem.prototype.addParticle.call(that.renderer);
+        ++that.totalParticles;
+    };
+
+    this.started = false;
+
+    var renderer = runtimeScene.getLayer("").getRenderer();
+    renderer.addRendererObject(this.renderer, runtimeObject.getZOrder());
+    this._convertYPosition = renderer.convertYPosition;
+};
+gdjs.ParticleEmitterObjectRenderer = gdjs.ParticleEmitterObjectCocosRenderer;
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.getRendererObject = function(){
+    return this.renderer;
+};
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.update = function(delta){
+    this.renderer.update(delta);
+    if(!this.started && this.getParticleCount() > 0){
+        this.started = true;
+    }
+};
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.setPosition = function(x, y){
+    this.renderer.setPosition(cc.p(x, this._convertYPosition(y)));
+};
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.setAngle = function(angle1, angle2){
+    this.renderer.setAngle(-(angle1 + angle2)/2.0);
+    this.renderer.setAngleVar(Math.abs(angle2 - angle1)/2.0);
+};
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.setForce = function(min, max){
+    this.renderer.setSpeed((min + max)/2.0);
+    this.renderer.setSpeedVar(Math.abs(max - min)/2.0);
+};
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.setZoneRadius = function(radius){
+    this.renderer.setPosVar(cc.p(radius, radius));
+};
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.setLifeTime = function(min, max){
+    this.renderer.setLife((min + max)/2.0);
+    this.renderer.setLifeVar(Math.abs(max - min)/2.0);
+};
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.setGravity = function(x, y){
+    this.renderer.setGravity(cc.p(x, -y));
+};
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.setColor = function(r1, g1, b1, r2, g2, b2){
+    var a = this.renderer.getStartColor().a;
+    this.renderer.setStartColor(cc.color(r1, g1, b1, a));
+    a = this.renderer.getEndColor().a;
+    this.renderer.setEndColor(cc.color(r2, g2, b2, a));
+};
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.setAlpha = function(alpha1, alpha2){
+    var color = this.renderer.getStartColor();
+    color.a = alpha1;
+    this.renderer.setStartColor(color);
+    color = this.renderer.getEndColor();
+    color.a = alpha2;
+    this.renderer.setEndColor(color);
+};
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.setSize = function(size1, size2){
+    this.renderer.setStartSize(2*this.originalSize*size1/100.0);
+    this.renderer.setEndSize(2*this.originalSize*size2/100.0);
+};
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.setFlow = function(flow){
+    this.renderer.setEmissionRate(flow);
+};
+
+gdjs.ParticleEmitterObjectCocosRenderer.prototype.validTexture = function(texture, runtimeScene){
```

Comment:
As it's returning a boolean, it's better to name this function `isTextureValid` :)